sensum-ethereum/test/sensations-test.js

```diff
@@ -1,20 +1,32 @@
 const { expect } = require("chai");
 const { ethers } = require("hardhat");
+const legacySensations = require("./legacy-sensations.json");
 
-// FIXME: Use Sensations contract!
-describe("Greeter", function () {
-  it("Should return the new greeting once it's changed", async function () {
-    const Greeter = await ethers.getContractFactory("Greeter");
-    const greeter = await Greeter.deploy("Hello, world!");
-    await greeter.deployed();
+const checkSensation = async (contract, sensation, expectedLength) => {
+  const tx = await contract.newSensation(sensation);
+  const receipt = await tx.wait();
+  const eventArgs = receipt.events.find(({ event }) => event === "Synapsis")
+    .args[0];
 
-    expect(await greeter.greet()).to.equal("Hello, world!");
+  expect(await contract.getSensationsLength()).to.equal(expectedLength);
+  expect(eventArgs[1]).to.equal(sensation.message);
+  expect(parseInt(eventArgs[0]._hex, 16)).to.equal(sensation.avatar);
+};
 
-    const setGreetingTx = await greeter.setGreeting("Hola, mundo!");
-
-    // wait until the transaction is mined
-    await setGreetingTx.wait();
-
-    expect(await greeter.greet()).to.equal("Hola, mundo!");
+describe("Sensum", function () {
```

Comment:
memo: A test about a sensation failing given is it "too large" it would be cool too!